### PR TITLE
Upgrade Joukowski model to handle rated conditions

### PR DESCRIPTION
**Pull request by lawrenceccheung**
Addresses issue in #784.

I updated the Joukowski model to handle rated conditions, following the formulation in Sorensen (2022): 
https://wes.copernicus.org/preprints/wes-2022-108/

Note: this pull request is in draft state, please do not merge until we run tests to confirm that the Joukowski model behaves properly at rated Ct.

**EDIT**: I added some results, let's review it and see if it's OK to use.

Lawrence

### CODE REVIEW

**Comment by lawrenceccheung** [comment]
I was able to test out the new Joukowski model on a couple of turbines, and able to get them to behave better at rated wind speeds.

For the 2.4-107 model:
image

For the 1.79-100
image

However, please note a couple of things:
- I'm interpreting the $C_{T,r}$ variable (`Ct_rated`) to mean the value of Ct when it's operating in region 2, and not the value of Ct when the turbine hits rated power.   That way the solid body term vanishes in region 2 and power production is unaffected in that part of the power curve:

$S_0 = \alpha(\frac{C_{T,r}-C_T}{C_{T,r}})^\beta$

- I'm using a different value of $\alpha$ compared to Sorensen's paper.  While I don't expect the same parameters to hold across all turbine models, it turns out I had to use a _negative_ $\alpha$ in order to limit the power above rated.  I'm using $\alpha=-0.21$ for $C_T > C_{T,r}$, whereas Sorensen used $\alpha=0.05$.  I'm not sure if there's a typo in the paper or a mistake somewhere in the code.  Or could it also be related to the fact that we switched signs on the rotor rotation direction (https://github.com/Exawind/amr-wind/pull/848)?

Lawrence

**Comment by lawrenceccheung** [comment]
@psakievich It occurred to me that we could also put in a generator efficiency and directly report electrical power generation.   Let me do that in a separate PR (maybe long with some documentation) after this one is approved and merged.

Lawrence

File: amr-wind/wind_energy/actuator/disk/Joukowsky.H [hunk 1]
```diff
@@ -22,6 +22,10 @@ struct JoukowskyData : public DiskBaseData
     // --- Sorenson 2020 equation 10 constants ----
     amrex::Real root_correction_coefficient{2.335};
     amrex::Real root_correction_exponent{4.0};
+    // --- Sorenson 2022 equation 26 constants ----
```
Comment (psakievich) [inline amr-wind/wind_energy/actuator/disk/Joukowsky.H:25]: @lawrenceccheung let's add a comment here about the change in sign, and using 0.2 vs 0.8 for the C_t < C_t,r case.

File: amr-wind/wind_energy/actuator/disk/Joukowsky_ops.H [hunk 2]
```diff
@@ -180,20 +204,34 @@ struct ComputeForceOp<Joukowsky, ActSrcDisk>
...
+        const amrex::Real term1 = std::pow(a2 * lambda + a3 * S0, 2.0);
+        const amrex::Real term2 =
+            a1 * (0.5 * Ct - 2.0 * a4 * lambda * S0 - a5 * S0 * S0);
+        const amrex::Real term3 = a2 * lambda + a3 * S0;
 
         const amrex::Real numerator = std::sqrt(term1 + term2) - term3;
 
         const amrex::Real denominator =
-            amrex::max<amrex::Real>(4.0 * a1, machine_eps);
+            amrex::max<amrex::Real>(1.0 * a1, machine_eps);
```
Comment (psakievich) [inline amr-wind/wind_energy/actuator/disk/Joukowsky_ops.H:234]: A little pedantic, but we can just remove the coefficient here now.

**Review by psakievich — APPROVED** [review]
A couple of minor things to clean up but nothing to stop the merge if required.

File: amr-wind/wind_energy/actuator/disk/Joukowsky.H (hunk 1 above)
Comment (lawrenceccheung) [reply amr-wind/wind_energy/actuator/disk/Joukowsky.H]: Thanks @psakievich, I added some comments explaining those values in the latest commit.

**Pull request merged**